src/stdio_mgr/types.py

```diff
@@ -0,0 +1,175 @@
+r"""``stdio_mgr.types`` *code module*.
+
+``stdio_mgr.types`` provides context managers for convenient
+interaction with ``stdin``/``stdout``/``stderr`` as a tuple.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+import collections.abc
+from contextlib import suppress
+from io import TextIOBase
+
+from stdio_mgr.compat import AbstractContextManager
+
+
+class MultiItemTuple(collections.abc.Iterable):
```

Comment:
I am not wedded to keeping MultiItemTuple - I suspect that there are better ways to achieve this - or ClosingStdioTuple which is unlikely to be useful as we would be using _MultiCloseContextManager or variants.

Reply:
I think the next thing I'm going to do is get that inheritance diagram set up in the Sphinx docs, and get RTD set up.  That should hopefully really help me see how things flow.

Reply:
@jayvdb First pass. Renders smaller than I'd like, but zooming in the webpage view works ok.

A [template change](https://stackoverflow.com/a/57840173/4376000) should allow it to expand, if desired.